src/d3a/blockchain/contracts/ink/lib.rs

```diff
@@ -0,0 +1,203 @@
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract(version = "0.1.0")]
+mod erc20 {
```

Comment:
Could rename to something as simple as `Trade` to get an idea right from the start, what this contract is trying to achieve.

Reply:
Good point, I will delete the erc20 code snippets and rename to trade.wasm for now